src/views/TimezoneModal.vue

```diff
@@ -2,44 +2,60 @@
   <ion-header>
     <ion-toolbar>
       <ion-buttons slot="start">
-        <ion-button @click="closeModal"> 
+        <ion-button @click="closeModal">
           <ion-icon :icon="close" />
         </ion-button>
       </ion-buttons>
       <ion-title>{{ $t("Select time zone") }}</ion-title>
     </ion-toolbar>
     <ion-toolbar>
-      <ion-searchbar @ionFocus="selectSearchBarText($event)" :placeholder="$t('Search time zones')"  v-model="queryString" @keyup.enter="queryString = $event.target.value; findTimeZone()" @keydown="preventSpecialCharacters($event)" />
+      <ion-searchbar @ionFocus="selectSearchBarText($event)" :placeholder="$t('Search time zones')" v-model="queryString"
+        @keyup.enter="queryString = $event.target.value; findTimeZone()" @keydown="preventSpecialCharacters($event)" />
     </ion-toolbar>
   </ion-header>
-
+ 
+ 
   <ion-content class="ion-padding">
     <!-- Empty state -->
-    <div class="empty-state" v-if="filteredTimeZones.length === 0">
-      <p>{{ $t("No time zone found")}}</p>
+    <div>
+      <div class="empty-state" v-if="loading && timeZones.length === 0">
+        <ion-item lines="none" color="none">
+          
+          <ion-spinner class="center-spinner" color="secondary" name="crescent" slot="start" ></ion-spinner>
+          {{ $t("Fetching time zones") }}
+        </ion-item>
+      </div>
     </div>
-
+    <div v-if="!loading && filteredTimeZones.length === 0">
+      <div class="empty-state">
```

Comment:
No need to have extra div for adding if condition, adding a class and if condition can be handled in a single div.